src/software/ai/navigator/path_planner/theta_star_path_planner.h

```diff
@@ -297,26 +316,48 @@ class ThetaStarPathPlanner : public PathPlanner
     void resetAndInitializeMemberVariables(const Rectangle &navigable_area,
                                            const std::vector<ObstaclePtr> &obstacles);
 
-    // if close to destination then return no path
+    /**
+     * Returns a collision-free hash for the given Coordinate
+     *
+     * @param coord Coordinate to hash
+     *
+     * @return hash of the Coordinate
+     */
+    unsigned long hashCoordinate(const Coordinate &coord) const;
+
+    /**
+     * Returns a collision-free hash for the given a pair of Coordinate
+     *
+     * @param coord1 Coordinate 1 of pair to hash
+     * @param coord2 Coordinate 2 of pair to hash
+     *
+     * @return hash of the pair of Coordinates
+     */
+    unsigned long hashCoordinatePair(
+        const ThetaStarPathPlanner::Coordinate &coord1,
+        const ThetaStarPathPlanner::Coordinate &coord2) const;
+
+    // if close to end then return no path
     static constexpr double CLOSE_TO_DEST_THRESHOLD = 0.01;  // in metres
 
     // increase in threshold to reduce oscillation
-    static constexpr int BLOCKED_DESINATION_OSCILLATION_MITIGATION =
+    static constexpr unsigned int BLOCKED_DESINATION_OSCILLATION_MITIGATION =
```

Comment:
confused about this constant. It's a factor to scale `CLOSE_TO_DEST_THRESHOLD` to prevent oscillations? I'm not sure how? (Haven't read implementation at this point).

Is the `CLOSE_TO_DEST_THRESHOLD` even super necessary? Maybe we could remove both constants?

Reply:
I think I was observing oscillations due to overshoot so these constants control that